sos/plugins/python.py

```diff
@@ -42,4 +48,41 @@ def setup(self):
         else:
             self.add_cmd_output('python -V', suggest_filename='python-version')
 
+        if self.get_option('gather_py_hashes'):
+            digests = {
+                'digests': []
+            }
+
+            py_paths = [
+                '/usr/lib',
+                '/usr/lib64',
+                '/usr/local/lib',
+                '/usr/local/lib64'
+            ]
+
+            for py_path in py_paths:
+                for py_file in Path(py_path).glob("**/*.py"):
+                    try:
+                        with open(str(py_file), 'rb') as f:
+                            digest = hashlib.sha256()
+                            chunk = 1024
+                            while True:
+                                data = f.read(chunk)
+                                if data:
+                                    digest.update(data)
+                                else:
+                                    break
+
+                            digest = digest.hexdigest()
+
+                            digests['digests'].append({
+                                'filepath': str(py_file),
+                                'sha256': digest
+                            })
+                    except IOError:
+                        self._log_error(
+                            "Unable to read python file at %s" % py_file)
+
+            self.add_string_as_file(json.dumps(digests), 'digests.json')
```

Comment:
This will store the `digest.json` inside `sos_strings/python/digests.json` - is that intentional? Usually, plugins collect data either via `add_copy_spec` (to copy directly into sosreport's directory structure) or `add_cmd_output` (to collect output under `sos_commands/python`) - those two locations (esp. the `sos_commands/python`) are the first places where one would search for `python` plugin output.

`add_tring_as_file` sounds a good usage here (you dont collect a file or a cmd output here), just the whole idea of:

- collecting filenames and hashes of *any* custom python scripts
- storing them on bit "unusual" location

this sounds like a Big Brother approach to me.

Btw. isn't it sufficient to e.g.:
- run something like `rpm -V $(rpm -qa | grep python)` to verify no changes in installed python packages (which basically should cover all files you are interested in, am I right?)
- files not reported by the `rpm -V` match original files in the packages (so you can get their hashes from the packages themselves, if needed)
- for files reported by the `rpm -V`, we can run the checksums

This approach sounds less intrusive as it skips custom applications, while it - at the end - provides sufficient information for your purposes.

Reply:
@pmoravec I could not find a way how I can store the generated .json file into sosreport's directory. Of course, I would like to store it there, however, `add_string_as_file` was the only way I could find to store my .json file.

Reply:
This isn't that different from what the `unpackaged` plugin does, and using `add_string_as_file()` is the correct way to store in-memory string data generated by a plugin in the report. It's much less common than `add_copy_spec()`, but that method should only be used to collect actual file system locations for inclusion in the report.